ERCS/erc-7679.md

```diff
@@ -0,0 +1,223 @@
+
+### UserOp builder
+
+To conform to this standard, a smart account implementation must provide a “UserOp builder” contract that implements the IUserOperationBuilder interface, as defined below:
+
+
+```solidity
+struct Execution {
+    address target;
+    uint256 value;
+    bytes callData;
+}
+
+interface IUserOperationBuilder {
+		/**
+     * @dev Returns the ERC-4337 EntryPoint that the account implementation
+     * supports.
+     */
+    function entryPoint() external view returns (address);
+    
+    /**
+     * @dev Returns the nonce to use for the UserOp, given the context.
+     * @param smartAccount is the address of the UserOp sender.
+     * @param context is the data required for the UserOp builder to
+     * properly compute the requested field for the UserOp.
+     */
+    function getNonce(
+        address smartAccount,
+        bytes calldata context
+    ) external view returns (uint256);
+	
+    /**
+     * @dev Returns the calldata for the UserOp, given the context and
+     * the executions.
+     * @param smartAccount is the address of the UserOp sender.
+     * @param executions are (destination, value, callData) tuples that
+     * the UserOp wants to execute.  It's an array so the UserOp can
+     * batch executions.
+     * @param context is the data required for the UserOp builder to
+     * properly compute the requested field for the UserOp. 
+     */
+    function getCallData(
+        address smartAccount,
+        Execution[] calldata executions,
+        bytes calldata context
+    ) external view returns (bytes memory);
+    
+    /**
+     * @dev Returns the dummy signature for the UserOp, given the context
+     * and the executions.
+     * @param smartAccount is the address of the UserOp sender.
+     * @param executions are (destination, value, callData) tuples that
+     * the UserOp wants to execute.  It's an array so the UserOp can
+     * batch executions.
+     * @param context is the data required for the UserOp builder to
+     * properly compute the requested field for the UserOp.
+     */
+    function getDummySignature(
+        address smartAccount,
+        Execution[] calldata executions,
+        bytes calldata context
+    ) external view returns (bytes memory signature);
+    
+    /**
+     * @dev Returns a correctly encoded signature, given a UserOp that
+     * has been correctly filled out except for the signature field.
+     * @param smartAccount is the address of the UserOp sender.
+     * @param userOperation is the UserOp.  Every field of the UserOp should
+     * be valid except for the signature field.  The "PackedUserOperation"
+     * struct is as defined in ERC-4337.
+     * @param context is the data required for the UserOp builder to
+     * properly compute the requested field for the UserOp.
+     */
+    function getSignature(
```

Comment:
It is highly dangerous to trust an external entity (rpc node) to create a signature. 
You can use view function to generate the hash to sign (similar to getUserOpHash), and maybe "embedsig" if you need to populate the "signature" field with other data in addition to the signature itself. 
But never trust an external entity to hold private information and generate a signature.